src/data-publication/reports/tables/Aggregate2.jsx

```diff
@@ -24,15 +25,52 @@ const renderValues = (values, key) => {
   return values.map((value, index) => {
     return [
       <td key={`count-${key}-${index}`}>{value.count}</td>,
-      <td key={`value-${key}-${index}`}>{value.value}</td>
+      <td key={`value-${key}-${index}`}>{value.value}</td>,
     ]
   })
 }
 
-const Aggregate2 = React.forwardRef((props, ref) => {
-  if (!props.report) return null
+/**
+ * Reducer function to aggregate Disposition data by dispositionName
+ * @param {Object} prev
+ * @param {Object} curr
+ * @returns Array[Object]
+ */
+ const valueReducer = (prev, curr) => {
+  // Initialize tracker object for this dispositionName if necessary
+  if (!prev[curr.dispositionName])
+    prev[curr.dispositionName] = {
+      dispositionName: curr.dispositionName,
+      value: 0,
+      count: 0,
+    }
+
+  // Summarize content
+  prev[curr.dispositionName].count += curr.count
+  prev[curr.dispositionName].value += curr.value
+  return prev
+}
+
+/**
+ * Temporary Fix: Identifies unaggregated Disposition entries and aggregates them
+ * @param {Array} data
+ */
+const fixUnaggregatedValue = data => {
+  const idxsNeedAggregation = []
+
+  data.forEach((x, idx) => x.values.length > 7 && idxsNeedAggregation.push(idx))
+
+  // Aggregate un-aggregated value data
+  idxsNeedAggregation.forEach(i => {
+    const obj = data[i].values.reduce(valueReducer, {})
+    data[i].values = Object.keys(obj).map(k => obj[k])
+  })
+
+  return data
+}
 
-  const sortedTracts = props.report.tracts.sort(function(tractA, tractB) {
+export const sortAndFix = (report) => {
```

Comment:
Second case where we are using `sortAndFix` function.

Reply:
The `Fix` is different between Agg1 and Agg2 but we can refactor as mentioned above.